core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -58,8 +67,16 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   protected abstract Map<String, String> getProperties();
 
   @Override
-  public void prepareRun(BatchSinkContext context) {
-    PluginProperties pluginProperties = context.getPluginProperties();
-    context.addOutput(pluginProperties.getProperties().get(Properties.BatchReadableWritable.NAME));
+  public void prepareRun(BatchSinkContext context) throws DatasetManagementException {
+    Map<String, String> properties = getProperties();
+    if (pluginConfig.containsMacro(Properties.BatchReadableWritable.NAME)) {
+      if (!context.datasetExists(properties.get(Properties.BatchReadableWritable.NAME))) {
```

Comment:
rather than doing this, I think it would be better to have a BatchReadableWritableConfig with a single macro enabled name field that all the other configs extend.